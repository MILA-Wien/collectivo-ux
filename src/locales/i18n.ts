import { nextTick } from "vue";
import { createI18n } from "vue-i18n";

export const SUPPORT_LOCALES = ["en", "de"];

export function setupI18n(
  options = {
    locale: "en",
    legacy: false,
    fallbackWarn: false,
    missingWarn: false,
  }
) {
  const i18n = createI18n(options);
  setI18nLanguage(i18n, options.locale);
  return i18n;
}

export default setupI18n();

export function setI18nLanguage(i18n: any, locale: any) {
  if (i18n.mode === "legacy") {
    i18n.global.locale = locale;
  } else {
    i18n.global.locale.value = locale;
  }
}

export async function loadLocaleMessages(i18n: any, locale: any) {
  // load locale messages with dynamic import
  import(`./locales/${locale}.json`)
    .catch((e) => {
      console.log("error", e, locale);
    })
    .then((messages) => {
      if (messages) {
        // set locale and locale message
        i18n.global.setLocaleMessage(locale, messages.default);

        return nextTick();
      }
    });
}
