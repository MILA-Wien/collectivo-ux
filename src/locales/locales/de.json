{
    "menu.About": "Über",
    "menu.Login": "Login",
    "menu.Menu": "Menu",
    "Log out": "Logout",
    "This is an about page": "Dies ist eine Über-Seite",
    "Counter": "Zähler",
    "Increment": "Hinzufügen",
    "My membership": "Meine Mitgliedschaft",
    "Membership": "Mitgliedschaft",
    "Member data": "Mitgliedsdaten",
    "Loading members": "Lade Mitglieder",
    "Showing":"Zeige",
    "to": "bis",
    "of": "von",
    "Edit user":"Benutzer bearbeiten",
    "Cancel":"Abbrechen",
    "Save":"Speichern",
    "Members":"Mitglieder",
    "members":"Mitglieder",
    "Do nothing":"Nichts tun",
    "Id":"Id",
    "admin_attr":"Admin Attribute",
    "user_attr":"Benutzer Attribute",
    "actions":"Aktionen",
    "user_id":"Benutzer Id",
    "tile.members_registration_tile":"Registrierung",
    "Edit": "Bearbeiten",
    "Registration": "Registrierung",
    "Next": "Weiter",
    "Back": "Zurück",
    "Main menu": "Hauptmenü"
}