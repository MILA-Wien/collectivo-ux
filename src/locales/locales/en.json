{
    "menu.About":"About",
    "This is an about page":"This is an about page",
    "Counter":"Counter",
    "Increment":"Increment"


}