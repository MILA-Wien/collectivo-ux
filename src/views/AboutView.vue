<template>
  <div class="about">
    <h1>{{$t("This is an about page")}}</h1>
    <br />
    <div>{{$t("Counter")}}: {{ store.count }}</div>
    <button @click="increment">{{$t("Increment")}}</button>
  </div>
</template>
<script setup lang="ts">
import { useCounterStore } from "@/stores/counter";
const store = useCounterStore();
store.count++;
// with autocompletion âœ¨
store.$patch({ count: store.count + 1 });
// or using an action instead
store.increment();
const increment = () => {
  store.$patch({ count: store.count + 1 });
};

</script>
<style>
@media (min-width: 1024px) {
  .about {
    min-height: 100vh;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
  }
}
</style>
