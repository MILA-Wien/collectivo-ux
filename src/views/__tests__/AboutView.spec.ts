import { mount } from "@vue/test-utils";
import { createPinia } from "pinia";
import { test, expect } from "vitest";
import AboutView from "../AboutView.vue";
import { setupI18n, loadLocaleMessages } from "@/locales/i18n";

test("mount component", async () => {
  expect(AboutView).toBeTruthy();
  const i18n = setupI18n();
  const wrapper = mount(AboutView, {
    global: {
      plugins: [createPinia(), i18n], // initializes Pinia
      stubs: { AboutView },
    },
  });
  loadLocaleMessages(i18n, "en");

  expect(wrapper.text()).toContain("3");
  expect(wrapper.html()).toMatchSnapshot();

  await wrapper.get("button").trigger("click");

  expect(wrapper.text()).toContain("4");

  await wrapper.get("button").trigger("click");

  expect(wrapper.text()).toContain("5");
});
