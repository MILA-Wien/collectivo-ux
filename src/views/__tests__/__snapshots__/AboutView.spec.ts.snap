// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"about\\">
  <h1>This is an about page</h1><br>
  <div>Counter: 3</div><button>Increment</button>
</div>"
`;
